seed/seed20.go

```diff
@@ -518,13 +526,34 @@ func (s *seed20) loadEssentialMeta(filterEssential func(*asserts.ModelSnap) bool
 	return nil
 }
 
+func modesInclude(modes []string, mode string) bool {
+	if strutil.ListContains(modes, mode) {
+		return true
+	}
+	if mode == "run" {
+		// not ephemeral
+		return false
+	}
+	return strutil.ListContains(modes, "ephemeral")
```

Comment:
This is a bit hard to understand. Is it the same if we change the `if` condition to
```
if mode != "ephemeral" {
    return false
}
```
? And if it's something that can be explained in a couple of lines, a comment on why the "ephemeral" mode is treated specially would help. :-)

Reply:
AFAIU ephemeral is kind of special as it's a catch all for all non-run modes, so the snap can either specify an specific mode eg. "install", or "ephemeral" in which matches "install", "recovery", "factory-reset" etc.

Reply:
Maybe it could use a comment, something line:
```
	// the snap mentions explicit modes in which case the run mode would be listed
	// or has an implicit run mode
	if strutil.ListContains(modes, mode) {
		return true
	}
	if mode == "run" {
		// not ephemeral
		return false
	}
	// the snap can use a catch-all ephemeral mode for all non-run modes
	return strutil.ListContains(modes, "ephemeral")
}
```

Reply:
Yes, comments here would be good, otherwise it's not clear why 'mode' argument is "irrelevant" for the last case.